frontend/app-development/contexts/SettingsModalContext.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { SettingsModalContextProvider, useSettingsModalContext } from './SettingsModalContext';
+
+describe('SettingsModalContext', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
```

Comment:
As far as I understand, it's unnecessary to have clearAllMocks when mocks are defined within each test and its respective it/test block. 🤔

Reply:
Yep, I agree. I have removed this.